src/Microsoft.VisualStudio.Services.Agent/SecretMask.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    public interface ISecretMask
```

Comment:
They are "maskers", if that helps.
Re: internal implementation details... if AddRegex/AddValue/AddVariableName are extension methods (or something) that add RegexMask/ValueMask/VariableValueMask under the hood, then I agree.  As long as you're not proposing a switch statement :)